frontend/src/components/core/Sidebar/Sidebar.test.tsx

```diff
@@ -137,6 +142,33 @@ describe("Sidebar Component", () => {
     )
   })
 
+  it("does not render the SidebarNav component if 0 appPages", () => {
+    const wrapper = renderSideBar({
+      appPages: [],
+    })
+
+    expect(wrapper.find(SidebarNav).exists()).toBe(false)
+  })
+
+  it("does not render the SidebarNav component if 1 appPages", () => {
+    const wrapper = renderSideBar({
+      appPages: [{ pageName: "streamlit_app", pageScriptHash: "page_hash" }],
+    })
+
+    expect(wrapper.find(SidebarNav).exists()).toBe(false)
+  })
+
+  it("does render the SidebarNav component if 2 appPages", () => {
+    const wrapper = renderSideBar({
+      appPages: [
+        { pageName: "streamlit_app", pageScriptHash: "page_hash" },
+        { pageName: "streamlit_app2", pageScriptHash: "page_hash2" },
+      ],
+    })
+
+    expect(wrapper.find(SidebarNav).exists()).toBe(true)
+  })
+
```

Comment:
Since the logic for `shouldNavCollapse` function and resulting `collapseNav` prop passed to the SidebarNav lives in the Sidebar component, I think it would be good to test the expected outcomes. For example:
1. pageQuantity < 7 results in `collapseNav` prop  false
2. no sidebar elements results in false
3. user preference expanded (probably have to mock it) results in false
4. pageQuantity of 7, has sidebar elements, user pref collapse results in true

Reply:
Added them in [98d8011](https://github.com/streamlit/streamlit/pull/6050/commits/98d8011883089ae54f686b6fc66f77c88fd55c0b)